plasmapy/particles/particle_class.py

```diff
@@ -2023,9 +2023,11 @@ class CustomParticle(AbstractPhysicalParticle):
     charge : ~astropy.units.Quantity or ~numbers.Real, optional
         The electric charge of the custom particle.  If provided as a
         `~astropy.units.Quantity`, then it must be in units of electric
-        charge.  If provided as a real number, then it is treated as the
-        ratio of the charge to the elementary charge. Defaults to |nan|
-        C.
+        charge. Defaults to |nan| C.
+
+    Z : ~numbers.Real, optional, |keyword-only|
+        The :term:`charge number`, which is equal to the ratio of the
+        charge to the elementary charge.
```

Comment:
So, wait, the only difference between the two is that Z can take floats, but charge can take quantities? Otherwise these look the same, conceptually, for the end user. Are we sure this separation is necessary?

Reply:
I'm probably not remembering some earlier discussion...

Reply:
Right now there's potential for ambiguity when `charge` is given a real number.  In a lot of our functionality, if an argument expected to be a `Quantity` is instead a real number, it will often be assumed that the real number is in SI units (along with a warning).  Separating `Z` from `charge` will remove that ambiguity.  

We use `Z` (or in some places, `Z_mean`) as the charge number pretty much everywhere else, so having `Z` mean charge number here would make it a lot more consistent with the rest of our functionality.

I'm also finding it more convenient when working with `particle_input` and the particle factory function in #1867 to have `CustomParticle` accept `Z` and have it mean the same thing as for a `Particle`.  

Anyway, these are my thoughts for this!

Reply:
Right, thanks for the reminders! 😅 That does make sense.